apps/api/src/purchase-orders/purchase-orders.service.ts

```diff
@@ -122,9 +131,60 @@ export class PurchaseOrdersService {
         totalCost: fullDto.totalCost,
         orderedDate: fullDto.orderDate,
         expectedDeliveryDate: fullDto.expectedDeliveryDate,
+        createdBy: user.email,
+        approvers: approvers.map((x) => x.email),
       },
     });
 
     return fullDto;
   }
+
+  async updateApprovalStatus(
+    id: number,
+    status: string,
+  ): Promise<PurchaseOrderFullDto> {
+    const oldOrder = await this.prisma.purchaseOrder.findUnique({
+      where: {id},
+      select: {
+        status: true,
+      },
+    });
+
+    if (!oldOrder) return null;
+
+    const purchaseOrder = await this.prisma.purchaseOrder.update({
+      where: {id},
+      data: {
+        status,
+      },
+      include: {
+        createdBy: true,
+        lineItems: {
+          include: {
+            item: {
+              include: {
+                parentItem: true,
+              },
+            },
+          },
+        },
+      },
+    });
+
+    if (!purchaseOrder) {
+      return null;
+    }
+
+    this.eventPublisher.publish<PurchaseOrderStatusChangedEvent>({
+      eventType: 'procurement.purchase-order-status-changed',
+      data: {
+        purchaseOrderId: purchaseOrder.id,
+        createdBy: purchaseOrder.createdBy.email,
+        oldStatus: oldOrder.status,
+        newStatus: purchaseOrder.status,
+      },
+    });
```

Comment:
Publishes a new event whenever the status changes. This gives the consumer a chance to send emails like "Your PO was approved", among other things.